src/main/java/application/rest/v1/actions/AppPodlistFunction.java

```diff
@@ -0,0 +1,162 @@
+ * limitations under the License.
+ */
+
+package application.rest.v1.actions;
+
+import java.util.List;
+
+import com.google.gson.JsonArray;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+
+import application.rest.v1.ComponentInfoRegistry;
+import application.rest.v1.KAppNavEndpoint;
+import application.rest.v1.Selector;
+import application.rest.v1.ComponentKind;
+import io.kubernetes.client.ApiClient;
+import io.kubernetes.client.ApiException;
+
+// apppodlist() or apppodlist(<application-namespace>,<application-name>)
+// Returns a list of pod names from all deployments belonging to specified application.
+// Return value structure is: { "pods" : "[<name1>,<name2>,etc]" }
+public class AppPodlistFunction implements Function {
+    
+    private static final String PODS_PROPERTY_NAME = "pods";
+    private static final String DEPLOYMENT_KIND = "Deployment";
+    private static final String APPLICATION_KIND = "Application";
+    private static final String POD_KIND = "Pod";
+    private static final String METADATA_PROPERTY_NAME = "metadata";
+
+    @Override
+    public String getName() {
+        return "apppodlist";
+    }
+
+    @Override
+    public boolean allowedParameterCount(int parameterCount) {
+        return parameterCount == 0 || parameterCount == 2;
+    }
+
+    @Override
+    public String invoke(ResolutionContext context, List<String> parameters) {
+        
+        final ApiClient client = context.getApiClient();
+        final ComponentInfoRegistry registry = context.getComponentInfoRegistry();
+        final JsonObject resource; 
+        String result = null;     
+
+        if (parameters.size() == 0) {
+            if (!APPLICATION_KIND.equals(context.getResourceKind())) {
+                // The context resource isn't a deployment.
+                return null;
+            }
+            resource = context.getResource();
+        }
+        // Two parameter function
+        else {
+            // parameters.get(0) :: appNamespace
+            // parameters.get(1) :: appName
+            final String appNamespace = parameters.get(0);
+            final String appName = parameters.get(1);
+            try {
+                //get application namespaced object
+                Object o = registry.getNamespacedObject(client, APPLICATION_KIND, appNamespace, appName);
+                resource = KAppNavEndpoint.getItemAsObject(client, o);      
+                result = getPodListFromApp(client, resource, registry);         
+            }
+            catch (ApiException e) {
+                return null;
+            }  
+        }
+        return result;                   
+    }
+
+
+    private String getPodListFromApp(ApiClient client, JsonObject resource, ComponentInfoRegistry registry) {
+        PodlistResult result = new PodlistResult();       
+        final Selector aSelector = Selector.getSelector(resource);                
```

Comment:
Need to check whether Deployment is one of the application's component kinds. If and only if Deployment is one of the component kinds does the selector match deployments.

Reply:
I am not sure if it is needed to check Deployment is one of the application's component kind as I did it earlier and I didn't see it makes any difference with checking or without checking since I already get Deployment kind object from app.   I still add it.

Reply:
If Deployment isn't in the component kind list then deployments aren't part of the app. The selector could still potentially match a deployment somewhere in the cluster, but since it isn't one of the component kinds it isn't part of the app.